src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php

```diff
@@ -714,6 +720,107 @@ private function getMergeFormData(int $orderId, int $shipmentId): array
         ];
     }
 
+    #[AdminSecurity("is_granted('update', 'AdminOrders')", redirectRoute: 'admin_orders_view', redirectQueryParamsToKeep: ['orderId'], message: 'You do not have permission to edit this.')]
+    public function splitShipmentAction(int $orderId, int $shipmentId, Request $request): RedirectResponse
+    {
+        $data = $request->get('split_shipment', []);
+        $carrier = isset($data['carrier']) ? (int) $data['carrier'] : null;
+
+        $productsRaw = $data['products'] ?? [];
+        $products = [];
+
+        foreach ($productsRaw as $prod) {
+            if (isset($prod['selected']) && (int) $prod['selected'] === 1) {
+                $products[] = [
+                    'id_order_detail' => (int) $prod['order_detail_id'],
+                    'quantity' => (int) $prod['selected_quantity'],
+                ];
+            }
+        }
+
+        $this->dispatchQuery(new SplitShipment($shipmentId, $products, $carrier));
+
+        return $this->redirectToRoute('admin_orders_view', [
+            'orderId' => $orderId,
+            'shipmentId' => $shipmentId,
+        ]);
+    }
+
+    /**
+     * @throws CoreException
+     * @throws Exception
+     */
+    #[AdminSecurity("is_granted('update', 'AdminOrders')", message: 'You do not have permission to show this.')]
+    public function getShipmentSplitForm(
+        Request $request,
+        #[Autowire(service: 'PrestaShop\PrestaShop\Adapter\Order\Repository\OrderDetailRepository')] OrderDetailRepository $orderDetailRepository,
+    ): Response {
+        $orderId = (int) $request->query->get('orderId');
+        $shipmentId = (int) $request->query->get('shipmentId');
+        $productsFromQuery = $request->get('products', []);
+        $selectedCarrier = (int) $request->query->get('carrier');
+
+        $orderShipmentProducts = $this->mergeProductsFromQueries($shipmentId, $productsFromQuery, $orderDetailRepository);
+
+        $splitShipmentTypeForm = $this->createForm(SplitShipmentType::class, [
+            'products' => $orderShipmentProducts,
+            'carrier' => $selectedCarrier,
+            'shipment_id' => $shipmentId,
+            'order_id' => $orderId,
+        ]);
+
+        return $this->render('@PrestaShop/Admin/Sell/Order/Order/Blocks/View/split_shipment_form.html.twig', [
+            'splitShipmentForm' => $splitShipmentTypeForm->createView(),
+            'orderId' => $orderId,
+            'shipmentId' => $shipmentId,
+        ]);
+    }
+
+    /**
+     * @param int $shipmentId
+     * @param array $productsFromQuery
```

Comment:
```suggestion
     * @param array $productsFromQuery
```

Can we add more details on what is expected?